app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.html

```diff
@@ -35,7 +35,32 @@
     <div class="band band-info" ng-if="isTemplatedPipeline && !pipeline.locked">
       <span class="glyphicon glyphicon small glyphicon-lock"></span> Manual edits are not allowed on templated pipelines
     </div>
-
+    <div class="band band-info" ng-if="hasDynamicSource && pipelineExecutions.length">
+      <span class="glyphicon glyphicon small glyphicon-wrench"></span> This template has a dynamic source. The
+      configuration is currently rendered using
+      <span ng-if="pipelineExecutions.length > 1" class="dropdown" uib-dropdown>
+        &nbsp;&nbsp;<button type="button" class="btn btn-xs btn-default dropdown-toggle" uib-dropdown-toggle>
+          <execution-build-title execution="currentExecution" default-to-timestamp="true" ></execution-build-title> <span class="caret"></span>
+        </button>
+        <ul class="dropdown-menu" role="menu" uib-dropdown-menu>
+          <li ng-repeat="p in pipelineExecutions" ng-class="{disabled: p.id == currentExecution.id}">
+            <a title="{{p.startTime | timestamp}}" ui-sref="{ executionId: p.id }"><execution-build-title execution="p" default-to-timestamp="true" ></execution-build-title>
+              <span class="text-muted small">({{p.startTime | relativeTime}})</span>
+              <span class="glyphicon small glyphicon-ok" ng-if="p.id == currentExecution.id"></span>
+            </a>
+          </li>
+        </ul>
+      </span>
+      <span ng-if="pipelineExecutions.length === 1"><execution-build-title execution="currentExecution" default-to-timestamp="true" ></execution-build-title></span>
+    </div>
+    <div class="band band-warning" ng-if="pipelineExecutions && !pipelineExecutions.length">
+      <span class="glyphicon glyphicon small glyphicon-alert"></span> This template has a dynamic source. The
+      configuration cannot be rendered before the pipeline has executed at least once.
+    </div>
+    <div class="band band-warning" ng-if="templateError && pipelineExecutions.length && !pipeline.isNew">
```

Comment:
The conditions here aren't quite complete, because you can hit a template planning error even if you're not using dynamic sources - and now if you're not using dynamic sources, those errors are going to be silent.

In the past, template planning errors were surfaced in the configuration modal (and if you hit a planning error when loading the pipeline config view, the template configuration modal opened automatically). I'd like to stick to that model.

Reply:
Nothing has really changed if you're not configuring a pipeline with dynamic source. This will always be false if you're not using dynamic sources; and the modal will open like it did before before.
The modal is not of much help when an error occurs while using dynamic sources, because it looks like this:
![Configuration modal](https://user-images.githubusercontent.com/155558/32366474-827a6fac-c07e-11e7-8ef4-5f5bd98f03a0.png)